mist-common/src/main/java/edu/snu/mist/common/operators/ApplyStatefulOperator.java

```diff
@@ -79,30 +75,30 @@ public void processLeftData(final MistDataEvent input) {
 
     input.setValue(output);
     outputEmitter.emitData(input);
+    latestTimestampBeforeCheckpoint = input.getTimestamp();
   }
 
   @Override
   public void processLeftWatermark(final MistWatermarkEvent input) {
     outputEmitter.emitWatermark(input);
     if (input.isCheckpoint()) {
-      latestCheckpointTimestamp = input.getTimestamp();
+      final Map<String, Object> stateMap = new HashMap<>();
+      stateMap.put("applyStatefulFunctionState", new Cloner().deepClone(applyStatefulFunction.getCurrentState()));
+      checkpointMap.put(input.getTimestamp(), stateMap);
```

Comment:
Maybe we use latestTimestampBeforeCheckpoint as a key because we want to replay event processing after this timestamp

Reply:
Then, we dont need to use the checkpoint timestamp

Reply:
You're right. I will change this.